app/ide-desktop/lib/client/src/index.ts

```diff
@@ -225,9 +230,29 @@ class App {
 
     /** Start the backend processes. */
     async startBackendIfEnabled() {
-        await this.runIfEnabled(this.args.options.engine, () => {
+        await this.runIfEnabled(this.args.options.engine, async () => {
+            // The first return value is the original string, which is not needed.
+            // These all cannot be null as the format is known at runtime.
+            const [, projectManagerHost, projectManagerPort] =
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                GLOBAL_CONFIG.projectManagerEndpoint.match(/^ws:\/\/(.+):(.+)$/)!
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            this.projectManagerHost ??= projectManagerHost!
+            this.projectManagerPort ??= await portfinder.getPortPromise({
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                port: parseInt(projectManagerPort!),
+            })
+            const projectManagerUrl = `ws://${this.projectManagerHost}:${this.projectManagerPort}`
+            this.args.groups.engine.options.projectManagerUrl.value = projectManagerUrl
             const backendOpts = this.args.groups.debug.options.verbose.value ? ['-vv'] : []
-            projectManager.spawn(this.args, backendOpts)
+            const backendEnv = Object.assign({}, process.env, {
+                // These are environment variables, and MUST be in CONSTANT_CASE.
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                SERVER_HOST: this.projectManagerHost,
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                SERVER_PORT: `${this.projectManagerPort}`,
```

Comment:
This code prepares the `SERVER_HOST` and `SERVER_PORT` environment variables for the `project-manager`.